docs/howto/danceable-lyrics-input-tool/DanceableLyrics/backend/ayx_plugins/danceable_lyrics.py

```diff
@@ -26,13 +29,35 @@
 class DanceableLyrics(PluginV2):
     """Concrete implementation of an AyxPlugin."""
 
+    @staticmethod
+    def _validate_datasets_dir(datasets_dir: Path) -> None:
+        if not datasets_dir.is_dir():
+            raise WorkflowRuntimeError("Bad path")
+
+        filenames = [
+            "artists.csv",
+            "tracks.csv",
+            "r_track_artist.csv",
+            "genius_song_lyrics.csv",
+            "audio_features.csv"
+        ]
+        nonexistent_files = list(
+            filter(lambda filepath: not filepath.is_file(), map(lambda filename: (datasets_dir / filename), filenames))
+        )
+
+        if len(nonexistent_files) != 0:
```

Comment:
So another way of writing this instead of looping through the iterators twice is:
nonexistent_files = ', '.join(nonexistent_files); // ',' vs ', ' pending on how you want to space list of files
```
"file1, file2, file3, file4, file5"
```
If iter returns nothing, the result is an empty string.